dev/aarch64_clean/src/arith_native_aarch64.h

```diff
@@ -31,23 +31,63 @@ extern const int16_t mlk_aarch64_zetas_mulcache_twisted_native[];
 extern const uint8_t mlk_rej_uniform_table[];
 
 #define mlk_ntt_asm MLK_NAMESPACE(ntt_asm)
-void mlk_ntt_asm(int16_t *, const int16_t *, const int16_t *);
+void mlk_ntt_asm(int16_t *p, const int16_t *twiddles12345,
+                 const int16_t *twiddles56)
+/* This must be kept in sync with the HOL-Light specification
+ * in proofs/hol_light/arm/proofs/mlkem_ntt.ml */
+__contract__(
+  requires(memory_no_alias(p, sizeof(int16_t) * MLKEM_N))
+  requires(array_abs_bound(p, 0, MLKEM_N, 8192))
+  requires(twiddles12345 == mlk_aarch64_ntt_zetas_layer12345)
+  requires(twiddles56 == mlk_aarch64_ntt_zetas_layer67)
+  assigns(memory_slice(p, sizeof(int16_t) * MLKEM_N))
+  /* check-magic: off */
+  ensures(array_abs_bound(p, 0, MLKEM_N, 23595))
+  /* check-magic: on */
+);
 
 #define mlk_intt_asm MLK_NAMESPACE(intt_asm)
-void mlk_intt_asm(int16_t *, const int16_t *, const int16_t *);
+void mlk_intt_asm(int16_t *p, const int16_t *twiddles12345,
+                  const int16_t *twiddles56)
+/* This must be kept in sync with the HOL-Light specification
+ * in proofs/hol_light/arm/proofs/mlkem_intt.ml */
+__contract__(
+  requires(memory_no_alias(p, sizeof(int16_t) * MLKEM_N))
+  requires(twiddles12345 == mlk_aarch64_invntt_zetas_layer12345)
+  requires(twiddles56 == mlk_aarch64_invntt_zetas_layer67)
+  assigns(memory_slice(p, sizeof(int16_t) * MLKEM_N))
+  /* check-magic: off */
+  ensures(array_abs_bound(p, 0, MLKEM_N, 26625))
+  /* check-magic: on */
+);
 
 #define mlk_poly_reduce_asm MLK_NAMESPACE(poly_reduce_asm)
 void mlk_poly_reduce_asm(int16_t *);
 
 #define mlk_poly_tomont_asm MLK_NAMESPACE(poly_tomont_asm)
-void mlk_poly_tomont_asm(int16_t *);
+void mlk_poly_tomont_asm(int16_t *p)
+/* This must be kept in sync with the HOL-Light specification
+ * in proofs/hol_light/arm/proofs/mlkem_poly_tomont.ml */
+__contract__(
+  requires(memory_no_alias(p, sizeof(int16_t) * MLKEM_N))
+  assigns(memory_slice(p, sizeof(int16_t) * MLKEM_N))
+  ensures(array_abs_bound(p, 0, MLKEM_N, MLKEM_Q))
+);
 
 #define mlk_poly_mulcache_compute_asm MLK_NAMESPACE(poly_mulcache_compute_asm)
 void mlk_poly_mulcache_compute_asm(int16_t *, const int16_t *, const int16_t *,
                                    const int16_t *);
 
 #define mlk_poly_tobytes_asm MLK_NAMESPACE(poly_tobytes_asm)
-void mlk_poly_tobytes_asm(uint8_t *r, const int16_t *a);
+void mlk_poly_tobytes_asm(uint8_t *r, const int16_t *a)
+/* This must be kept in sync with the HOL-Light specification
+ * in proofs/hol_light/arm/proofs/mlkem_poly_tobytes.ml */
+__contract__(
+  requires(memory_no_alias(r, MLKEM_POLYBYTES))
+  requires(memory_no_alias(a, sizeof(int16_t) * MLKEM_N))
+  requires(array_bound(a, 0, MLKEM_N, 0, MLKEM_UINT12_LIMIT))
```

Comment:
This is still a mismatch from the HOL-Light spec. It's stronger, though, so let's leave it for now.